### ISSUE TITLE
Handle @Suppress("UNCHECKED_CAST") for UnsafeCast rule

### ISSUE THREAD

**Issue by xgouchet** [arturbosch/detekt#726]
IntelliJ / Android Studio detect when a cast is unsafe and show a warning, which can be disabled using the `@Suppress("UNCHECKED_CAST")` annotation. It seems redundant to also add the `UnsafeCast` rule, ending with some code annotated with `@Suppress("UNCHECKED_CAST", "UnsafeCast")`

**Comment by Mauin** [comment]
If the IntelliJ warning is enough for you, disabling the rule in your detekt configuration sounds like a good idea.

**Comment by xgouchet** [comment]
Unfortunately that doesn't do it for me. IntelliJ mostly gives a warning when generics are involved, from what I gathered. The following code won't trigger any warning in IntelliJ, but are detected by the UnsafeCast rule : 

```
val foo = something as AnotherThing
```

**Comment by Mauin** [comment]
Ah, I see. So its more about making sure the detekt rules also match the suppression rules of IntelliJ. We could add a List to each rule that can take in the names of "similar" rules in other static analysis suites. That way we could make this possible that only one `@Suppress` annotation will also suppress the detekt rule.

**Comment by xgouchet** [comment]
Yep that'd be great !

**Comment by Mauin** [comment]
I can take a look at implementing something that will make that work. WDYT @arturbosch @schalkms does something like this make sense?

**Comment by schalkms** [comment]
I makes sense. I think detekt has some rules where this mechanism can be applied. This means that this should be a **generic** functionality if possible.

**Comment by Mauin** [comment]
Yeah. I'll look into it later today/tomorrow

**Comment by arturbosch** [comment]
Yep, surprisingly I thought about the same thing yesterday 👍 for this
